.github/workflows/delete-caches.yml

```diff
@@ -20,23 +20,46 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v4.1.2
 
-      - name: Cleanup Branch Caches
+      - name: Setup gh-actions-cache extension
+        run: gh extension install actions/gh-actions-cache
+
+      - name: Retrieve Trunk SHA
+        id: get-sha
         run: |
-          gh extension install actions/gh-actions-cache
-          
           REPO=${{ github.repository }}
-          BRANCH=refs/pull/${{ github.event.pull_request.number }}/merge
-
-          echo "Fetching list of cache key"
-          cacheKeysForPR=$(gh actions-cache list -R $REPO -B $BRANCH | cut -f 1 )
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          SHA=$(gh pr view -R $REPO $PR_NUMBER --json mergeCommit --jq .mergeCommit.oid)
+          echo "sha=$SHA" >> $GITHUB_OUTPUT
 
-          ## Setting this to not fail the workflow while deleting cache keys. 
+      - name: Cleanup Caches
+        run: |
           set +e
-          echo "Deleting caches..."
-          for cacheKey in $cacheKeysForPR
-          do
-              gh actions-cache delete $cacheKey -R $REPO -B $BRANCH --confirm
+
+          REPO=${{ github.repository }}
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          PR_BRANCH=refs/pull/$PR_NUMBER/merge
+          
+          echo "Fetching list of cache keys for the PR branch"
+          cacheKeysForPR=$(gh actions-cache list -R $REPO -B $PR_BRANCH | cut -f 1)
+
+          echo "Deleting caches for PR branch..."
+          for cacheKey in $cacheKeysForPR; do
+              gh actions-cache delete $cacheKey -R $REPO -B $PR_BRANCH --confirm
           done
```

Comment:
Not blocking but might be worth considering adding any GitHub context vars like `${{ ... }}` to an env block in the step like:

```
env:
  REPO: ${{ github.repository }}
```

This makes it easier just to copy/paste the entire step in bash and shellcheck it (since it will all be valid) and makes it more obvious what its inputs are.

And then just reference "${REPO}" as an env var in the bash. Also, total nit but for casing bash vars, I'd change `cacheKeysForPR` to `cache_keys_for_pr`.

Reply:
Good suggestions. Added in 89f4d07adf2105c4bb9f8ecfeed51acf052a5826)